### ISSUE TITLE
NaN ELBO values with LKJcorr prior

### ISSUE THREAD

**Issue by sachinruk** [pymc-devs/pymc#1852]
Part of my model looks like as shown below where `alpha` and `beta` are regression parameters across `n_dep_id` items. When I have that LKJ prior all my ELBO values are either `-inf` or `nan`s. If I change it to the line that is commented out instead and remove this prior it starts getting proper numbers for ELBOs. To be completely fair though the numbers are still large negative values (eg. -2,145,119.88).

I can't see anything obviously wrong here. If it helps `dim = 210`. Is there any reason that ADVI would clash with these correlation matrices and cause these nans?

```
dim = n_dep_id
n_elem = dim * (dim - 1) / 2
tri_index = np.zeros([dim, dim], dtype=int)
tri_index[np.triu_indices(dim, k=1)] = np.arange(n_elem)
tri_index[np.triu_indices(dim, k=1)[::-1]] = np.arange(n_elem)

with pm.Model() as model:
    sigma_alpha = pm.Uniform('sigma_alpha', lower=0, upper=1, shape=n_dep_id)
    sigma_beta = pm.Uniform('sigma_beta', lower=0, upper=1, shape=n_dep_id)
    
    corr_triangle_u = pm.LKJCorr(
        'corr_u', n=100, p=dim, testval=np.random.randn(n_elem))
    corr_matrix_u = corr_triangle_u[tri_index]
    corr_matrix_u = T.fill_diagonal(corr_matrix_u, 1)
#     corr_matrix_u = T.eye(n_dep_id)
    cov_matrix_alpha = T.diag(sigma_alpha).dot(corr_matrix_u.dot(T.diag(sigma_alpha)))
    cov_matrix_beta = T.diag(sigma_beta).dot(corr_matrix_u.dot(T.diag(sigma_beta)))
```

**Comment by aseyboldt** [comment]
Why `n=100`? Shouldn't that put a bit much mass near the identity matrix? But I think the problem is your testval. Why do you set that in the first place? Those values do not correspond to a positive definite matrix. Some eigenvalues are negative, and pymc3 fails when computing the log det. Does it work if you leave that out?
A parametrisation based on the cholesky decomposition of the corr matrix wouldn't have this problem, but this does not seem to be implemented in pymc3 yet.

**Comment by sachinruk** [comment]
So I set `n=100` on purpose in the hope that it would be the same or similar to setting `corr_matrix_u = T.eye(n_dep_id)`. I did take out the `testval` without much luck. Changed `n=1` and still nothing, ELBOs are still `nan`.

**Comment by aseyboldt** [comment]
I can't reproduce the nans (I'm on master), but I do get -infs. The support of the entries of the correlation matrix is off. Hopefully this will be fixed by #1863.
